File: core/src/main/java/io/github/shaksternano/borgar/core/media/graphics/drawable/CompositeDrawable.java
package io.github.shaksternano.borgar.core.media.graphics.drawable;

/**
 * A drawable that is composed of multiple drawables. The way the drawables
 * are combined when drawn is up to the implementation of this interface.
 */
public interface CompositeDrawable extends Drawable {

    /**
     * Adds a part to the composite drawable.
     *
     * @param part The part to add.
     */
    void addPart(Drawable part);
}

File: core/src/main/kotlin/io/github/shaksternano/borgar/core/graphics/drawable/CompositeDrawable.kt
package io.github.shaksternano.borgar.core.graphics.drawable

import io.github.shaksternano.borgar.core.io.closeAll

abstract class CompositeDrawable : Drawable {

    val parts: MutableList<Drawable> = mutableListOf()

    override suspend fun close() =
        closeAll(parts)
}
